src/v/rpc/simple_protocol.cc

```diff
@@ -99,61 +99,66 @@ simple_protocol::dispatch_method_once(header h, net::server::resources rs) {
-                  // but we will not give a rseponse.
-                  rpclog.debug("Timing out request on gate_closed_exception "
-                               "(shutting down)");
-                  reply_buf.set_status(rpc::status::request_timeout);
-              } catch (...) {
-                  rpclog.error(
-                    "Service handler threw an exception: {}",
-                    std::current_exception());
-                  rs.probe().service_error();
-                  reply_buf.set_status(rpc::status::server_error);
-              }
-              return send_reply(ctx, std::move(reply_buf))
-                .finally([m, l = std::move(l)]() mutable {
-                    m->probes.latency_hist().record(std::move(l));
-                });
-          });
-    });
+            return m->handle(ctx->res.conn->input(), *ctx)
+              .then_wrapped([ctx, m, l = ctx->res.hist().auto_measure(), rs](
+                              ss::future<netbuf> fut) mutable {
+                  netbuf reply_buf;
+                  try {
+                      reply_buf = fut.get0();
+                      reply_buf.set_status(rpc::status::success);
+                  } catch (const rpc_internal_body_parsing_exception& e) {
+                      // We have to distinguish between exceptions thrown by the
+                      // service handler and the one caused by the corrupted
+                      // payload. Data corruption on the wire may lead to the
+                      // situation where connection is not longer usable and so
+                      // it have to be terminated.
+                      ctx->pr.set_exception(e);
+                      return ss::now();
+                  } catch (const ss::timed_out_error& e) {
+                      reply_buf.set_status(rpc::status::request_timeout);
+                  } catch (const ss::gate_closed_exception& e) {
+                      // gate_closed is typical during shutdown.  Treat
+                      // it like a timeout: request was not erroneous
+                      // but we will not give a rseponse.
+                      rpclog.debug(
+                        "Timing out request on gate_closed_exception "
+                        "(shutting down)");
+                      reply_buf.set_status(rpc::status::request_timeout);
+                  } catch (...) {
+                      rpclog.error(
+                        "Service handler threw an exception: {}",
+                        std::current_exception());
+                      rs.probe().service_error();
+                      reply_buf.set_status(rpc::status::server_error);
+                  }
+                  return send_reply(ctx, std::move(reply_buf))
+                    .finally([m, l = std::move(l)]() mutable {
+                        m->probes.latency_hist().record(std::move(l));
+                    });
+              });
+        });
+    } catch (const std::exception& e) {
```

Comment:
When making a change that re-indents a lot of lines, it is helpful to split it into two commits -- one making the functional change (but violating formatting rules), and then a second commit that reformats the code.

In my workflow, that means switching off 'reformat on save' in my editor, doing my change, committing, and then running "task lint:cpp" to reformat the code, then committing again.

Reply:
+1 to what John said, especially in this case where the changes are small.

when changes are a bit more complex, it can often be worth the extra time to convert a function to use co-routines before making the change. these huge re-indents are usually because of the formatting that seastar continuation passing needs to be readable.

Reply:
Got it
Thanks!